students/ericstreit/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,217 @@
+        except Exception as e:
+            logging.error(f'Hm, we had an error here?')
+            logging.error(e)
+
+    print(f' the number of records in the products collection is: {db.products.count()}')
+    print(f' the number of records in the customers collection is: {db.customers.count()}')
+    print(f' the number of records in the rentals collection is: {db.rentals.count()}')
+    return (record_count, error_count)
+
+def show_available_products():
+    """
+    Returns a Python dictionary of products listed as available with
+    the following fields:
+
+    product_id.
+    description.
+    product_type.
+    quantity_available.
+    """
+    mongo = MongoDBConnection()
+
+    with mongo:
+        #connect to Mongo
+        db = mongo.connection.hp_norton
+
+        #name the first collection we will work on
+        products = db["products"]
+        customers = db["customers"]
+        rentals = db["rentals"]
+
+        for name in products.find():
+            logging.debug(f"{name.get('product_id')} : {name.get('quantity_available')}")
+            try:
+                if int(name.get('quantity_available')) > 0:
+                    print(f"{name.get('product_id')}: {name.get('description')} is available!")
+            except ValueError as e:
+                logging.debug(f"skipping {name.get('product_id')} due to error")
+                logging.error(e)
+
+
+def show_rentals(product_id):
+    """
+    Returns a Python dictionary with the following user information from
+    users that have rented products matching product_id:
+
+    user_id.
+    name.
+    address.
+    phone_number.
+    email.
+
+    :param arg1: The product_id to search for
+    """
+    # gosh, I'm repeating myself a lot here with these Mongo commands. could
+    # be simplified into its own function, yes??
+    mongo = MongoDBConnection()
+
+    with mongo:
+        #connect to Mongo
+        db = mongo.connection.hp_norton
+
+        #the collections we will be using
+        customers = db["customers"]
+        rentals = db["rentals"]
+        customer_list = []
+        customer_dict ={}
+        for rented_item in rentals.find():
+            logging.debug(f"Does {product_id} match {rented_item.get('product_id')}?")
+            try:
+                if rented_item.get('product_id') == product_id:
+                    logging.debug(f'found a match!')
+                    customer_list.append(rented_item.get('user_id'))
+                    logging.debug(f"Adding {rented_item.get('user_id')} to list")
+            except ValueError as e:
+                logging.debug(f"skipping {rented_item.get('product_id')} due to error")
+                logging.error(e)
+
+        for name in customers.find():
+            logging.debug(f"comparing found user id to the customer list")
+            logging.debug(f"looking for {name.get('user_id')} in {customer_list}")
+            try:
+                if name.get('user_id') in customer_list:
```

Comment:
Would it be more efficient to go through the customer list to see if the user_id in the customer list is in the database, rather than going through the database looking for each customer id in the customer_list? I feel like the customer_list is likely to be quite a bit smaller, so the loop will be smaller. Same for rented_items. You can use pymongo function find_one to just look for a particular record in the database.